plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/IcebergQueryRunner.java

```diff
@@ -46,64 +51,70 @@ public static DistributedQueryRunner createIcebergQueryRunner()
         return createIcebergQueryRunner(
                 Map.of(),
                 Map.of(),
-                List.of());
+                Optional.empty());
     }
 
     public static DistributedQueryRunner createIcebergQueryRunner(
             Map<String, String> extraProperties,
             Map<String, String> connectorProperties,
-            Iterable<TpchTable<?>> tables)
+            Optional<File> metastoreDirectory)
             throws Exception
     {
         return createIcebergQueryRunner(
                 extraProperties,
                 connectorProperties,
-                tables,
-                Optional.empty());
+                metastoreDirectory,
+                IcebergQueryRunner::initializeSchema);
```

Comment:
why not just `queryRunner -> queryRunner.execute("CREATE SCHEMA " + SCHEMA)` ?

Reply:
Original code inside `createIcebergQueryRunner` was:
```
queryRunner.execute("CREATE SCHEMA tpch");

copyTpchTables(queryRunner, "tpch", TINY_SCHEMA_NAME, session, tables);
```
Now it's extracted to method `initializeSchema` and called as default behaviour.
Also as it most typical https://github.com/trinodb/trino/pull/10894#discussion_r802741292 case I've made it a default behaviour to copy 2 most common tables.
In other usages initialiser is overloaded and behaviour is changed.

Reply:
@findepi actually i refactored this part again. Schema initialiser is now created with builder and is more generic + i think I've addressed all your concern. WDYT ?